<launch>

  <arg name="drone_id" default="0"/>

  <node pkg="swarm_bridge" name="drone_$(arg drone_id)_bridge_node_tcp" type="bridge_node_tcp" output="screen">
  
    <remap from="~my_odom" to="/drone_$(arg drone_id)/vins_fusion/imu_propagate"/>
    <remap from="~my_occupancy_inflate" to="/drone_$(arg drone_id)/ego_planner_node/grid_map/occupancy_inflate"/>
    <remap from="~my_goal_point" to="/drone_$(arg drone_id)/ego_planner_node/goal_point"/>
    <remap from="/takeoff_land_to_drones" to="/px4ctrl/takeoff_land"/>

    <param name="self_id" value="$(arg drone_id)"/>
    <param name="is_ground_station" value="false"/>
    <param name="odom_max_freq" value="10"/>
    <param name="occupancy_broadcast_freq" value="1"/>
    <param name="goal_point_freq" value="1"/>
    <param name="drone_num" value="16"/>
    <param name="drone_ip_0" value="192.168.1.21" type="string"/>
    <param name="drone_ip_1" value="192.168.50.22" type="string"/>
    <param name="drone_ip_2" value="192.168.50.23" type="string"/>
    <param name="drone_ip_3" value="192.168.50.24" type="string"/>
    <param name="drone_ip_4" value="192.168.50.31" type="string"/>
    <param name="drone_ip_5" value="192.168.50.32" type="string"/>
    <param name="drone_ip_6" value="192.168.50.33" type="string"/>
    <param name="drone_ip_7" value="192.168.50.34" type="string"/>
    <param name="drone_ip_8" value="192.168.50.35" type="string"/>
    <param name="drone_ip_9" value="192.168.50.36" type="string"/>
    <param name="drone_ip_10" value="192.168.50.37" type="string"/>
    <param name="drone_ip_11" value="192.168.50.38" type="string"/>
    <param name="drone_ip_12" value="192.168.50.39" type="string"/>
    <param name="drone_ip_13" value="192.168.50.40" type="string"/>
    <param name="drone_ip_14" value="192.168.50.41" type="string"/>
    <param name="drone_ip_15" value="192.168.50.42" type="string"/>
    <param name="ground_station_num" value="1"/>
    <param name="ground_station_ip_0" value="192.168.1.100" type="string"/>

    <param name="pub_occupancy" value="true" type="bool"/>

  </node>

  <!-- <node pkg="swarm_bridge" name="traj2odom_node" type="traj2odom_node" output="screen">

    <param name="odom_hz" value="30" />

  </node> -->

</launch>
